pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -352,12 +352,18 @@ func PublishDeviceInfoToZedCloud(ctx *zedagentContext) {
 		} else {
 			for _, zpool := range zpoolList {
 				rStorageInfo := new(info.StorageInfo)
+				rStorageInfo.ZpoolState = new(info.StorageState)
 				curentRaid := info.StorageRaidType_STORAGE_RAID_TYPE_NORAID
 				zpoolName := zpool.Properties[libzfs.PoolPropName].Value
+				storageState := zfs.GetZfsDeviceStatusFromStr(zpool.Properties[libzfs.PoolPropHealth].Value)
 				zpoolSizeInByte, err := strconv.ParseUint(zpool.Properties[libzfs.PoolPropSize].Value, 10, 64)
 				if err != nil {
 					log.Errorf("error with ParseUint for get zpool size in byte: %v", err)
 				}
+				countZvolume, err := zfs.GetZfsCountVolume(zpool.Properties[libzfs.PoolPropName].Value)
```

Comment:
Ditto about adding a check for nil zpool.Properties[libzfs.PoolPropName]

Reply:
I do not think that if we are on this line of code, the pool name will not be available to us or have nil.
But still. even if something goes wrong, we'll return 0 since we can't find a dataset for the nil value.

Reply:
It isn't about the pool name; it is about avoiding a potential golang panic and instead logging/reporting an error.